File: TASC-Intern-Batch9/pharmacy-backend-microservice/payment-service/src/main/java/com/pharmacy/payment_service/service/OrderService.java
package com.pharmacy.payment_service.service;

import com.pharmacy_backend.common.dto.response.ApiResponse;
import com.pharmacy_backend.common.dto.response.PageResponse;
import com.pharmacy_backend.order_service.dto.request.OrderFilterRequest;
import com.pharmacy_backend.order_service.dto.request.OrderRequest;
import com.pharmacy_backend.order_service.dto.response.OrderDetailResponse;
import com.pharmacy_backend.order_service.dto.response.OrderResponse;

import java.util.List;

public interface OrderService {
    ApiResponse<PageResponse<List<OrderResponse>>> getAllOrders(int pageIndex, int pageSize, OrderFilterRequest request);
    ApiResponse<PageResponse<List<OrderResponse>>> getMyOrders(int pageIndex, int pageSize, String status);
    ApiResponse<List<OrderDetailResponse>> getOrderDetail(Long orderId);
    ApiResponse<?> createOrder(OrderRequest request);
    ApiResponse<OrderResponse> changeOrderStatus(Long id, String status);
    ApiResponse<OrderResponse> changePaymentStatus(Long id, String paymentStatus);
    ApiResponse<Long> getTotalOrder();

    ApiResponse<Long> getAllRevenue();

    ApiResponse<List<OrderResponse>> getFiveNewestOrder();
    ApiResponse<Void> cancelOrder(Long id);
}

File: TASC-Intern-Batch9/fe/fe-client/src/api/orderService.ts
import axiosClient from './axiosClient';
import type { ApiResponse, PageResponse} from '../types';
import type { OrderRequest, OrderResponse, OrderDetailResponse } from '../types/order.types';


const orderService = {
  createOrder: (data: OrderRequest) => {
    return axiosClient.post<ApiResponse<{ paymentUrl?: string; orderId: number }>>('/orders', data);
  },

  getMyOrders: (pageIndex: number = 1, pageSize: number = 10, status?: string) => {
    const params: any = { pageIndex, pageSize };
    if (status) params.status = status;
    return axiosClient.get<ApiResponse<PageResponse<OrderResponse[]>>>('/orders/my-orders', { params });
  },

  getOrderDetail: (id: number) => {
    return axiosClient.get<ApiResponse<OrderDetailResponse[]>>(`/orders/detail/${id}`);
  },

  cancelOrder: (id: number) => {
    return axiosClient.put<ApiResponse<void>>(`/orders/cancel/${id}`);
  }
};

export default orderService;